.github/workflows/ci-gradle.yml

```diff
@@ -66,6 +66,7 @@ jobs:
         if: >
           inputs.publish_snapshots &&
           github.event_name != 'pull_request' &&
+          github.ref == 'refs/heads/main' &&
```

Comment:
The alternative would be to set the default of the `publish_snapshots` input to `false` and require this from the caller:

```yaml
    with:
      publish_snapshots: ${{ github.ref == 'refs/heads/main' }}
```

Reply:
That would be more work though. What are your thoughts here, @timtebeek ?

Reply:
I would probably be in favor of merging this one and then maybe do the other steps separately, if at all. Because this can really mess with things, as we've seen today and we don't even know if we maybe published something from a branch to SaaS production.